drivers/iio/light/cm36283.c

```diff
@@ -0,0 +1,891 @@
+		*val = chip->chip_info->als_it_times[2 * als_it_index + 0];
+		*val2 = chip->chip_info->als_it_times[2 * als_it_index + 1];
+		return IIO_VAL_INT_PLUS_MICRO;
+	case IIO_PROXIMITY:
+		switch (chip->chip_info->partid) {
+		case CM36283_ID:
+			ps_it_index = FIELD_GET(CM36283_PS_IT, chip->chip_info->ps_conf[CM36283_PS_CONF1]);
+			break;
+		case CM36686_ID:
+			ps_it_index = FIELD_GET(CM36686_PS_IT, chip->chip_info->ps_conf[CM36283_PS_CONF1]);
+			break;
+		default:
+			return -EINVAL;
+		}
+		*val = chip->chip_info->ps_it_times[2 * ps_it_index + 0];
+		*val2 = chip->chip_info->ps_it_times[2 * ps_it_index + 1];
+		return IIO_VAL_INT_PLUS_MICRO;
+	default:
+		return -EINVAL;
+	}
+}
+
+static int cm36283_write_light_int_time(struct cm36283_data *chip, int val2)
+{
+	struct i2c_client *client = chip->client;
+	int index = -1, ret, new_int_time;
+
+	for (int i = 0; i < chip->chip_info->num_als_it / 2; i++) {
+		if (chip->chip_info->als_it_times[2 * i + 1] == val2) {
+			index = i;
+			break;
+		}
+	}
+
+	if (index == -1)
+		return -EINVAL;
+
+	new_int_time = chip->chip_info->als_conf & ~CM36283_ALS_IT;
+	new_int_time |= FIELD_PREP(CM36283_ALS_IT, index);
+
+	ret = i2c_smbus_write_word_data(chip->client, CM36283_REG_ALS_CONF,
+					new_int_time);
+	if (ret < 0)
+		dev_err(&client->dev,
+			"Failed to set ALS integration time: %pe", ERR_PTR(ret));
+	else
+		chip->chip_info->als_conf = new_int_time;
+
+	return ret;
+}
+
+static int cm36283_write_prox_int_time(struct cm36283_data *chip, int val2)
+{
+	struct i2c_client *client = chip->client;
+	int index = -1, ret, new_int_time;
+
+	for (int i = 0; i < chip->chip_info->num_ps_it / 2; i++) {
+		if (chip->chip_info->ps_it_times[2 * i + 1] == val2) {
+			index = i;
+			break;
+		}
+	}
+
+	if (index == -1)
+		return -EINVAL;
+
+	switch (chip->chip_info->partid) {
+	case CM36283_ID:
+		new_int_time = chip->chip_info->ps_conf[CM36283_PS_CONF1] & ~CM36283_PS_IT;
+		new_int_time |= FIELD_PREP(CM36283_PS_IT, index);
+		break;
+	case CM36686_ID:
+		new_int_time = chip->chip_info->ps_conf[CM36283_PS_CONF1] & ~CM36686_PS_IT;
+		new_int_time |= FIELD_PREP(CM36686_PS_IT, index);
+		break;
+	default:
+		return -EINVAL;
+	}
+
+	ret = i2c_smbus_write_word_data(chip->client, CM36283_REG_PS_CONF1,
+					new_int_time);
+	if (ret < 0)
+		dev_err(&client->dev, "Failed to set PS integration time: %pe",
+			ERR_PTR(ret));
+	else
+		chip->chip_info->ps_conf[CM36283_PS_CONF1] = new_int_time;
+
+	return ret;
+}
+
+static int cm36283_read_raw(struct iio_dev *indio_dev,
+			    struct iio_chan_spec const *chan, int *val,
+			    int *val2, long mask)
+{
+	struct cm36283_data *chip = iio_priv(indio_dev);
+	int ret;
+
+	mutex_lock(&chip->lock);
+
+	switch (mask) {
+	case IIO_CHAN_INFO_RAW:
+		ret = cm36283_read_channel(chip, chan, val);
+		break;
+		case IIO_CHAN_INFO_INT_TIME:
+		ret = cm36283_read_int_time(chip, chan, val, val2);
+		break;
+	default:
+		return -EINVAL;
```

Comment:
```suggestion
		ret = -EINVAL;
```

To avoid issues with mutex.